src/storage/compression/bitpacking.cpp

```diff
@@ -36,21 +70,36 @@ struct BitpackingState {
 	idx_t total_size;
 	void *data_ptr;
 
+	BufferMinMax<T> compression_buffer_minmax; //! Min and Max values in the buffer;
+
 public:
+	void SubtractFrameOfReference(const T &frame_of_reference) {
+		for (idx_t i = 0; i < compression_buffer_idx; i++) {
+			compression_buffer[i] -= frame_of_reference;
+		}
+	}
+
 	template <class OP>
 	void Flush() {
-		bitpacking_width_t width = BitpackingPrimitives::MinimumBitWidth<T>(compression_buffer, compression_buffer_idx);
-		OP::Operation(compression_buffer, compression_buffer_validity, width, compression_buffer_idx, data_ptr);
-		total_size += (BITPACKING_WIDTH_GROUP_SIZE * width) / 8 + sizeof(bitpacking_width_t);
+		T frame_of_reference = compression_buffer_minmax.Minimum();
+		SubtractFrameOfReference(frame_of_reference);
+		bitpacking_width_t width = BitpackingPrimitives::MinimumBitWidth(compression_buffer, compression_buffer_idx);
```

Comment:
Can we gain an extra bit of compression here using the fact that even for signed values we can no longer have negative values here?

Reply:
Ah yea mean by providing the templated bool to MinimumBitWidth that we talked about?
Yes I'll see if I can get that to work

Reply:
We can optimize even further, as we don't need to do the additional loop that FindMinimumBitWidth is doing to get the minimum/maximum value, because we already keep track of that

Reply:
Hmm I don't know if I made a mistake, or this is again due to counting blocks instead of bytes.

The new theoretical compression ratio for `(2000 + i%2) for i in range(1024)` is 56.88
We're getting 41.0 now

Reply:
Thanks for explaining it to me, I think I understand the relation between row groups, vectors, blocks and segments a lot better now!